src/commitment.rs

```diff
@@ -4,10 +4,14 @@ use crate::polynomial;
 use crate::setup;
 use blst;
 
-pub struct Proof {}
+pub struct Proof {
+    pub qs_in_g1: blst::blst_p1,
+}
 
 pub struct Opening {
+    pub input: point::Point,
```

Comment:
i'm not super attached to the following but my thinking is that the verifier will know which `input` they are querying for along w/ the commitment ahead of time.

so the protocol flow goes something like:

- ahead of time, prover and verifier have the shared setup.
- prover publishes the commitment
- verifier sends over a query `x`.
- prover responds with the minimal information required which is the evaluation `y = f(x)` and the proof.
- verifier can verify `f(x) = y` with this information.

which means, if we want to keep the `Opening` struct minimal, e.g. because it will be serialized to disk/network, then it only contains `y` and the proof.

Reply:
Yep! Sounds good! I've updated the API accordingly